src/main/java/com/y/java_board/controller/ArticleController.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@Controller()
+public class ArticleController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ArticleService articleService;
+    private final CommentService commentService;
+
+    public ArticleController(ArticleService articleService, CommentService commentService){
+        this.articleService = articleService;
+        this.commentService = commentService;
+    }
+
+    @GetMapping("/articles")
+    public String articles(Model model){
+        List<Article> articles = articleService.findArticles();
+        model.addAttribute("articles", articles);
+        return "article/articles";
+    }
+
+    @GetMapping("/articles/new")
+    public String showCreateForm(Model model){
+        model.addAttribute("articleDto", new ArticleDto("","",""));
+        return "article/create";
+    }
+
+    @PostMapping("/articles/new")
+    public String create(ArticleDto articleDto, BindingResult result, Model model){
+        if(result.hasErrors()){
+            // model.addAttribute() 오류의 원인을 보낸다.
+            return "/articles/new";
+        }
+        articleService.createOne(articleDto.toEntity());
+        return "redirect:/articles";
+    }
+
+    @GetMapping("/articles/{id}")
+    public String showArticle(@PathVariable("id") long id, Model model){
+        Optional<Article> articleOptional = articleService.findOne(id);
+        if(articleOptional.isPresent()){
+            List<Comment> comments = commentService.findCommentsByArticleId(id);
+            model.addAttribute("article",articleOptional.get());
+            model.addAttribute("commentDto", new CommentDto("", "", id));
+            model.addAttribute("comments", comments);
+            return "/article/detail";
+        }
+        // TODO : 해당 id 가 없다는 안내가 필요할 것 같다.
+        return "redirect:/articles";
+    }
+
+    @GetMapping("/articles/delete/{id}")
+    public String deleteArticle(@PathVariable("id") long id, Model model){
+        Article article = articleService.findOne(id)
+                .orElseThrow(()-> new IllegalArgumentException("Invalid article Id : " + id));
+        articleService.deleteOne(id);
+        return "redirect:/articles";
+    }
+
+    @GetMapping("/articles/update/{id}")
+    public String showUpdateForm(@PathVariable long id, Model model){
+        Article article = articleService.findOne(id)
+                .orElseThrow(()-> new IllegalArgumentException("Invalid article Id: "+ id));
+        model.addAttribute("articleDto", new ArticleDto(article.getTitle(),article.getContent(), article.getWriter()));
+        return "/article/update";
+    }
+
+    @PostMapping("/articles/update/{id}")
```

Comment:
Do not have POST. Use PUT/PATCH Request

Reply:
Thanks to the [link](https://stackoverflow.com/questions/72744349/thymeleaf-thmethod-delete-put-leads-to-request-method-post-not-supported) you provided, I was able to use delete, put, and patch requests on the form as well. I had some inconveniences using only get and post, but I'm very satisfied after using the requests and mapping annotation.